src/fal/cli/selectors.py

```diff
@@ -234,6 +244,8 @@ def depth(self, selector: str) -> Optional[int]:
 OP_PARENTS = SelectorGraphOpDepth(re.compile("^(?P<depth>\\d*)\\+(?P<rest>.*)"))
 OP_CHILDREN = SelectorGraphOpDepth(re.compile("(?P<rest>.*)\\+(?P<depth>\\d*)$"))
 
+# Logic based set operators
+SET_INTERSECTION = SetOperator(",")
```

Comment:
Might be a silly question, but why do we need a class if we only instantiate it once?

Reply:
Right, I should have cleared it a bit more. Currently `fal` does not support using a space-delimited *single* string as an argument to `--select` (it interprets the whole thing [up to the `+`] as one):

```
$ temp_dir=/tmp/fal fal flow run --profiles-dir ../ --experimental-models --select "model_a+ model_c+"
...
networkx.exception.NetworkXError: The node model.fal_008.model_a+ model_c is not in the graph.
```

This is supported by `dbt run`, so I left an open door to extend this operation, and we could probably add a `SET_UNION` in the future with using this class to represent what each separator token (space => `set.union`, comma => `set.intersection`) would correspond to.

```
SET_UNION = SetOperator(" ", action=set.union)
SET_INTERSECTION = SetOperator(",", action=set.intersection)
```

Though I can simply do it in the simple way now, and change in the following PRs (if we decide to fix this bug). WDYT?

Reply:
Thanks for the clarification. I don't have a strong opinion either way, so, my vote would be to keep the class definition.

Reply:
So `argparse` does the `SetOperator(" ")` for us and we define `SetOperator(",")`. But how would you do the `action=...` behavior if we don't really have control on applying `SetOperator(" ")`. Would you `join` them?

Reply:
So `argparse` does the `SetOperator(" ")` for us and we define `SetOperator(",")`. But how would you do the `action=...` behavior if we don't really have control on applying `SetOperator(" ")`. Would you `join` them?

Reply:
> So argparse does the SetOperator(" ") for us and we define SetOperator(","). But how would you do the action=... behavior if we don't really have control on applying SetOperator(" "). Would you join them?

We have 2 different cases:
```bash
$ fal flow run --select foo bar
$ fal flow run --select "monty python" eggs "benedict,spam" 
```

Both of them are currently handled by the argparse, and they are parsed like this (by argparse):

```py
["foo", "bar"]
['"monty python", "eggs", '"benedict,spam"']
```

The first case looks perfectly fine, but on the second case (focusing on the initial result, `"monty python"`) we need a second round of splitting. We probably have 2 options:

A) (As you said) Join them all (since the actual argparse splitting behavior is necessary to capture all arguments after select) and split it afters
B) Iterate them, and split each individual item on that list once more. If there are no more spaces in them (like `eggs`), we'll do nothing. But if there are spaces in individual items, we'll split them by space and process them as usual (check for comma etc.)

The first method seems to be a bit complicated, due to the precedence problems (like which thing belongs to which group might be confusing). Though take all of these with a grain of salt, I haven't take a stab at the implementation of this problem yet. But in either way, I think the approach of using `SetOperator` works perfectly fine.